### PR TITLE
Add certification tests for MongoDB state store

### PR DESCRIPTION
**Pull request by yash-nisar**
Closes https://github.com/dapr/components-contrib/issues/1938

Signed-off-by: Yash Nisar <yashnisar@microsoft.com>

# Description

Added certification tests for MongoDB state store

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: https://github.com/dapr/components-contrib/issues/1938

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by berndverst — CHANGES_REQUESTED** [review]
Remove the `.DS_Store` file. This file's purpose is how the macOS finder application presents that folder to you. It serves no purpose for Dapr OSS. I opened a PR to add this to the gitignore file so it cannot happen again - Add macOS .DS_Store to gitignore

**Comment by yash-nisar** [comment]
Not sure how I missed that. Repushing.

**Review by berndverst — CHANGES_REQUESTED** [review]
Please do not modify anything at the `tests/certification` level.

You need to create a new submodule by adding a `go.mod` and `go.sum` in the `tests/certification/state/mongodb` folder.

Your changes here in `tests/certification/go.mod` and `tests/certification/go.sum` should be reverted.

**Comment by yash-nisar** [comment]
> Please do not modify anything at the `tests/certification` level.
> 
> You need to create a new submodule by adding a `go.mod` and `go.sum` in the `tests/certification/state/mongodb` folder.
> 
> Your changes here in `tests/certification/go.mod` and `tests/certification/go.sum` should be reverted.

Will do

File: tests/certification/state/mongodb/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+1. Able to create and test connection.
+2. Able to do set, fetch, update and delete.
+3. Negative test to fetch record with key, that is not present.
+
+## Component must reconnect when server or network errors are encountered
+
+## Infra test:
+1. When MongoDB goes down and then comes back up - client is able to connect
+
+## eTag related:
```
Comment (berndverst) [inline tests/certification/state/mongodb/README.md:17]: This is already covered by conformance test which has an eTag test suite. https://github.com/dapr/components-contrib/blob/master/tests/config/state/tests.yml#L6

File: tests/certification/state/mongodb/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+1. Insert a Key-Value pair, eTag will be 1  
+2. Update Value v2 for this Key with eTag equal to 1 - new eTag wil be 2.  
+3. Try to Update v3 for this Key with eTag equal to 4 - value should not get updated.  
+4. Confirm that the Value is v2 for this Key.  
+
+## Transaction related, like Upsert:
```
Comment (berndverst) [inline tests/certification/state/mongodb/README.md:23]: This is also already covered by conformance tests.
https://github.com/dapr/components-contrib/blob/master/tests/config/state/tests.yml#L6

File: tests/certification/state/mongodb/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# MongoDB State Store certification testing
+
+This project aims to test the MongoDB State Store component under various conditions.
+
+## Test plan
```
Comment (berndverst) [inline tests/certification/state/mongodb/README.md:5]: See https://docs.dapr.io/reference/components-reference/supported-state-stores/setup-mongodb/

Please add tests that verify the following metadata properties.

```
  - name: writeconcern
    value: <REPLACE-WITH-WRITE-CONCERN> # Optional.
  - name: readconcern
    value: <REPLACE-WITH-READ-CONCERN> # Optional.
  - name: operationTimeout
    value: <REPLACE-WITH-OPERATION-TIMEOUT> # Optional.
  - name: params
    value: <REPLACE-WITH-ADDITIONAL-PARAMETERS> # Optional.
```

For example the operationTimeout can probably be combined with the network interruption somehow.

**Review by berndverst — CHANGES_REQUESTED** [review]
I left several comments on the test plan. There are additional tests needed, and we do not need the Etag or Transaction test given that is covered by conformance tests already.

File: tests/certification/state/mongodb/README.md (hunk 1 above)
Comment (yash-nisar) [reply tests/certification/state/mongodb/README.md:17]: Fixed

File: tests/certification/state/mongodb/README.md (hunk 2 above)
Comment (yash-nisar) [reply tests/certification/state/mongodb/README.md:23]: Fixed

File: tests/certification/state/mongodb/README.md (hunk 3 above)
Comment (yash-nisar) [reply tests/certification/state/mongodb/README.md:5]: Fixed

**Review by berndverst — APPROVED** [review]

**Pull request merged**